pkg/lib/config/authentication_flow_bot_protection.go

```diff
@@ -41,7 +41,41 @@ var _ = Schema.Add("AuthenticationFlowBotProtectionProvider", `
 type AuthenticationFlowBotProtectionProvider struct {
 	Type BotProtectionProviderType `json:"type,omitempty"`
 }
+
+func (p *AuthenticationFlowBotProtectionProvider) Equal(p2 *AuthenticationFlowBotProtectionProvider) bool {
+	if p == nil && p2 == nil {
+		return true
+	}
+	if p == nil || p2 == nil {
+		return false
+	}
+
+	return p.Type == p2.Type
+}
```

Comment:
1. Modify the spec to remove `bot_protection.provider` in branch.
2. Conform to the spec.

Reply:
Modified spec in https://github.com/louischan-oursky/authgear-server/commit/36bfc95b33e603b28536df77be0bd48c9a56e0ac

Reply:
> conform to spec

Will depends this on #4550